packages/oauth/oauth-client/src/util.ts

```diff
@@ -116,3 +117,52 @@ export class CustomEventTarget<EventDetailMap extends Record<string, unknown>> {
     )
   }
 }
+
+export type SpaceSeparatedValue<Value extends string> =
+  | `${Value}`
+  | `${Value} ${string}`
+  | `${string} ${Value}`
+  | `${string} ${Value} ${string}`
+
+export const includesSpaceSeparatedValue = <Value extends string>(
+  input: string,
+  value: Value,
+): input is SpaceSeparatedValue<Value> => {
+  // Optimized version of:
+  // return input.split(' ').includes(value)
```

Comment:
Seems pretty nice!  Is there an issue with using this implementation?

Reply:
The `split()` implementation is [about 8 times slower](https://jsbench.me/s2m24q9yz6/1). I'm just tying to save the planet here, one CPU cycle at the time.